### PR TITLE
Update pre commit config

### PR DESCRIPTION
**Pull request by jgowdy**

### CODE REVIEW

File: docs/source/CONTRIBUTING.md [hunk 1]
```diff
@@ -109,9 +109,9 @@ to get used to the basic usage.
 
 Now, setting up a development environment is super simple! Additional info if you run into trouble: [Poetry Environments]
 
-Step 1: [Install Poetry]  
-Step 2: Run ``poetry install``  
-Step 3: Optionally Run ``poetry shell``  
+Step 1: [Install Poetry]
+Step 2: Run ``poetry install``
+Step 3: Optionally Run ``poetry shell``
```
Comment (tarkatronic) [inline docs/source/CONTRIBUTING.md]: Is there a way we can ignore Markdown files in the whitespace fixer? In Markdown, these are actually significant. Two spaces at the end of a line are used to indicate a line break, so removing these spaces breaks the formatting here.

Comment (tarkatronic) [reply docs/source/CONTRIBUTING.md]: If you want to see it locally, you can `poetry install -E docs` then `poetry run make -C docs html` and view `docs/build/html/CONTRIBUTING.html`

**Comment by jgowdy** [comment]
I will figure out how to filter trailing whitespace checker against .md files

**Review by tarkatronic — APPROVED** [review]

**Review by beardedeagle — APPROVED** [review]
LGTM

**Pull request merged**